plugin/trino-oracle/src/test/java/io/trino/plugin/oracle/BaseOracleConnectorTest.java

```diff
@@ -191,6 +195,119 @@ public void testTimestampOutOfPrecisionRounded()
         assertUpdate("DROP TABLE " + tableName);
     }
 
+    @Test
+    public void testCastProjectionPushdown()
+    {
+        for (CastTestCase testCase : ImmutableList.of(
+                new CastTestCase("char(10)", "char(20)"),
+                new CastTestCase("char(10)", "varchar(20)"),
+                new CastTestCase("char(501)", "varchar(20)"),
+                new CastTestCase("char(501)", "varchar"), // No cast will be used during execution, as both the type will be converted to nclob
+                new CastTestCase("varchar(10)", "varchar(20)"),
+                new CastTestCase("varchar(1001)", "varchar(20)"),
+                new CastTestCase("varchar(1001)", "varchar"), // No cast will be used during execution, as both the type will be converted to nclob
+                new CastTestCase("varchar", "varchar(20)"))) {
+            try (TestTable table = new TestTable(getQueryRunner()::execute, "test_cast_", "(a %s)".formatted(testCase.fromType()))) {
+                assertThat(query("SELECT CAST(a AS %s) FROM %s".formatted(testCase.castType(), table.getName())))
+                        .isFullyPushedDown();
```

Comment:
please add test for nested CAST:
`CAST(CAST(a AS %s) AS %s)`
https://github.com/trinodb/trino/pull/21939/commits/edc25bf5d855ec6b82cac562d2a5407570446db7#diff-01f7e5fb80c1ce75cf0ea5d3f1933156eb7984967b99b92090bf931e93569696R1055